quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/update/QuarkusCli38to315UpdateIT.java

```diff
@@ -0,0 +1,364 @@
+        expectedNewProperties.put("quarkus.package.jar.appcds.builder-image", "docker://dontknow");
+        expectedNewProperties.put("quarkus.package.jar.appcds.use-container", "false");
+        expectedNewProperties.put("quarkus.package.jar.compress", "false");
+        expectedNewProperties.put("quarkus.package.jar.filter-optional-dependencies", "false");
+        expectedNewProperties.put("quarkus.package.jar.add-runner-suffix", "false");
+        expectedNewProperties.put("quarkus.package.jar.user-configured-ignored-entries", "false");
+        expectedNewProperties.put("quarkus.package.jar.user-providers-directory", "true");
+        expectedNewProperties.put("quarkus.package.jar.included-optional-dependencies", "true");
+        expectedNewProperties.put("quarkus.package.jar.include-dependency-list", "true");
+        expectedNewProperties.put("quarkus.package.jar.decompiler.enabled", "true");
+        expectedNewProperties.put("quarkus.package.jar.decompiler.jar-directory", "/tmp");
+        expectedNewProperties.put("quarkus.package.jar.manifest.attributes.enabled", "true");
+        expectedNewProperties.put("quarkus.package.jar.manifest.sections.enabled", "true");
+        expectedNewProperties.put("quarkus.package.jar.manifest.add-implementation-entries", "true");
+
+        QuarkusCLIUtils.checkPropertiesUpdate(quarkusCLIAppManager, oldProperties, expectedNewProperties);
+    }
+
+    /**
+     * In quarkus 3.10, some dependencies are added if certain conditions are met.
+     * But in this case, no dependencies are removed, so cannot use .checkDependenciesUpdate().
+     * See https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.10#flyway-10-gear-white_check_mark
+     */
+    @Test
+    public void addDependenciesTest() throws XmlPullParserException, IOException {
+        List<Dependency> oldDependencies = new ArrayList<>();
+        oldDependencies.add(new QuarkusDependency("io.quarkus:quarkus-flyway"));
+        oldDependencies.add(new QuarkusDependency("io.quarkus:quarkus-jdbc-derby"));
+        oldDependencies.add(new QuarkusDependency("io.quarkus:quarkus-jdbc-db2"));
+        oldDependencies.add(new QuarkusDependency("io.quarkus:quarkus-jdbc-postgresql"));
+
+        List<Dependency> expectedDependencies = new ArrayList<>();
+        expectedDependencies.add(new QuarkusDependency("org.flywaydb:flyway-database-derby"));
+        expectedDependencies.add(new QuarkusDependency("org.flywaydb:flyway-database-db2"));
+        expectedDependencies.add(new QuarkusDependency("org.flywaydb:flyway-database-postgresql"));
+
+        QuarkusCliRestService app = quarkusCLIAppManager.createApplication();
+        addDependenciesToPom(app, oldDependencies);
+        quarkusCLIAppManager.updateApp(app);
+
+        List<Dependency> actualDependencies = getDependencies(app);
+        expectedDependencies.forEach((dependency) -> Assertions.assertTrue(actualDependencies.contains(dependency),
```

Comment:
In this case old dependenceis are not removed. Otherwise I would use `.checkDependenciesUpdate()` which wraps all of that logic.

BTW: It is in the comment for this method:
```
...
But in this case, no dependencies are removed, so cannot use .checkDependenciesUpdate().
...
```